src/managed/Microsoft.Extensions.DependencyModel/DependencyContextJsonReader.Netstandard2_0.cs

```diff
@@ -0,0 +1,628 @@
+                        path = reader.ReadAsString();
+                        break;
+                    case DependencyContextStrings.HashPathPropertyName:
+                        hashPath = reader.ReadAsString();
+                        break;
+                    case DependencyContextStrings.RuntimeStoreManifestPropertyName:
+                        runtimeStoreManifestName = reader.ReadAsString();
+                        break;
+                    default:
+                        reader.Skip();
+                        break;
+                }
+            }
+
+            reader.CheckEndObject();
+
+            return new LibraryStub()
+            {
+                Hash = hash,
+                Type = Pool(type),
+                Serviceable = serviceable,
+                Path = path,
+                HashPath = hashPath,
+                RuntimeStoreManifestName = runtimeStoreManifestName
+            };
+        }
+
+        private List<RuntimeFallbacks> ReadRuntimes(ref Utf8JsonReader reader)
+        {
+            var runtimeFallbacks = new List<RuntimeFallbacks>();
+
+            reader.ReadStartObject();
+
+            while (reader.Read() && IsTokenTypeProperty(reader.TokenType))
+            {
+                string runtime = reader.GetString();
+                var fallbacks = reader.ReadStringArray();
+
+                runtimeFallbacks.Add(new RuntimeFallbacks(runtime, fallbacks));
+            }
+
+            reader.CheckEndObject();
+
+            return runtimeFallbacks;
+        }
+
+        private static ArraySegment<byte> ReadToEnd(Stream stream)
+        {
+            int written = 0;
+            byte[] rented = null;
+
+            try
+            {
+                long expectedLength = 0;
+
+                if (stream.CanSeek)
+                {
+                    expectedLength = Math.Max(1L, stream.Length - stream.Position);
+                    rented = ArrayPool<byte>.Shared.Rent(checked((int)expectedLength));
+                }
+                else
+                {
+                    rented = ArrayPool<byte>.Shared.Rent(UnseekableStreamInitialRentSize);
+                }
+
+                int lastRead;
+
+                do
+                {
+                    if (expectedLength == 0 && rented.Length == written)
+                    {
+                        byte[] toReturn = rented;
+                        rented = ArrayPool<byte>.Shared.Rent(checked(toReturn.Length * 2));
+                        Buffer.BlockCopy(toReturn, 0, rented, 0, toReturn.Length);
+                        // Holds document content, clear it.
+                        ArrayPool<byte>.Shared.Return(toReturn, clearArray: true);
+                    }
+
+                    lastRead = stream.Read(rented, written, rented.Length - written);
+                    written += lastRead;
+                } while (lastRead > 0);
+
+                return new ArraySegment<byte>(rented, 0, written);
```

Comment:
Why are we reading the whole stream contents into an array and then parsing that, rather than reading and parsing as we go?  Is it because we expect these inputs to be rather small?

Reply:
> Why are we reading the whole stream contents into an array and then parsing that, rather than reading and parsing as we go?

Mainly because reading partially from the stream and passing that reader along the various methods is quite complex since the `Utf8JsonReader` doesn't have built-in support for streams. I will see if I can make it work but we need to provide APIs to support the stream scenario.

> Is it because we expect these inputs to be rather small?

I am not sure how big the input can get. Generally deps.json files are not that large (i.e. < 1MB). @eerhardt, what is the typical (and largest) size you expect as input to `DependencyContextJsonReader`?